runtime-modules/content/src/lib.rs

```diff
@@ -635,198 +658,484 @@ decl_module! {
 
             // Trigger event
             Self::deposit_event(RawEvent::CuratorRemoved(curator_group_id, curator_id));
-            Ok(())
         }
 
+        // TODO: Add Option<reward_account> to ChannelCreationParameters ?
         #[weight = 10_000_000] // TODO: adjust weight
         pub fn create_channel(
             origin,
-            owner: ChannelOwner<T::MemberId, T::CuratorGroupId, T::DAOId>,
-            params: ChannelCreationParameters<ContentParameters<T>>,
-        ) -> DispatchResult {
-            Ok(())
+            actor: ContentActor<T::CuratorGroupId, T::CuratorId, T::MemberId>,
+            params: ChannelCreationParameters<ContentParameters<T>, T::AccountId>,
+        ) {
+            ensure_actor_authorized_to_create_channel::<T>(
+                origin,
+                &actor,
+            )?;
+
+            // The channel owner will be..
+            let channel_owner = Self::actor_to_channel_owner(&actor)?;
+
+            // Pick out the assets to be uploaded to storage system
+            let content_parameters: Vec<ContentParameters<T>> = Self::pick_content_parameters_from_assets(&params.assets);
+
+            let expected_channel_id = NextChannelId::<T>::get();
+
+            let object_owner = StorageObjectOwner::<T>::Channel(expected_channel_id);
+
+            // check assets can be uploaded to storage.
+            // update can_add_content() to only take &refrences
+            T::StorageSystem::can_add_content(
+                object_owner.clone(),
+                content_parameters.clone(),
+            )?;
+
+            // TODO:
+            // Enough funds to create channel?
+            // Limit on number of channels per owner?
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // TODO: Burn funds from sender account as fee for creating a channel
+
+            // add assets to storage
+            // This should not fail because of prior can_add_content() check!
+            T::StorageSystem::atomically_add_content(
```

Comment:
Don't see the purpose of making the same check twice in a row, to be honest. Wouldn't it be better to remove the additional check inside of a `T::StorageSystem::atomically_add_content` method implementation?

Reply:
yes I see in this extrinsic the can_add_content() check is really redundant because there is nothing between it and the call to add_content(). It would have made sense when I had some additional checks there.

But no definitely keep the check inside atomically_add_content() it is safer. I'll remove the can_add_content check in this dispatch instead.

Reply:
done in a40b58fd87d2c645e95ffebc69c81c8501a09bb2